libyul/AssemblyStack.cpp

```diff
@@ -274,6 +274,44 @@ std::pair<MachineAssemblyObject, MachineAssemblyObject> AssemblyStack::assembleW
 	return {std::move(creationObject), std::move(deployedObject)};
 }
 
+std::pair<std::shared_ptr<evmasm::Assembly>, std::shared_ptr<evmasm::Assembly>> AssemblyStack::assembleEVM(optional<string_view> _deployName) const
```

Comment:
Why do we need a new function here?

Reply:
Why a string view?

Reply:
Note this function is a copy of `assembleWithDeployed`, which is just above this one in the file.

Reply:
So can we deduplicate the code?

Reply:
Yes, please read https://github.com/ethereum/solidity/pull/10866#issuecomment-824895496 :)

Reply:
Deduplicated.